processing/src/main/java/io/druid/segment/QueryableIndexIndexableAdapter.java

```diff
@@ -180,10 +181,12 @@ public int indexOf(String value)
           boolean done = false;
 
           {
-            dimensions = Maps.newLinkedHashMap();
+            final ArrayList<DictionaryEncodedColumn> dictionaryEncodedColumnArrayList = new ArrayList<>();
+
             for (String dim : getDimensionNames()) {
-              dimensions.put(dim, input.getColumn(dim).getDictionaryEncoding());
+              dictionaryEncodedColumnArrayList.add(input.getColumn(dim).getDictionaryEncoding());
             }
+            this.dictionaryEncodedColumns = dictionaryEncodedColumnArrayList.toArray(new DictionaryEncodedColumn[dictionaryEncodedColumnArrayList.size()]);
```

Comment:
it might be more readable to do something like
```java
FluentIterable.from(getDimensionNames())
        .transform(<function dim -> column>)
        .toArray()
```